source/fundamentals/crud/write-operations/bulk-write.txt

```diff
@@ -0,0 +1,552 @@
+.. _csharp-bulk-write:
+
+=====================
+Bulk Write Operations
+=====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: insert, update, replace, code example
+
+Overview
+--------
+
+Consider a scenario in which you want to insert a document into a collection,
+update multiple other documents, then delete a document. If you use
+individual methods, each operation requires its own database call. This guide
+shows you how to use bulk write operations to perform multiple write operations
+in a single database call.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<csharp-quickstart>` tutorial.
+
+Define the Write Operations
+---------------------------
+
+For each write operation you want to perform, create an instance of one of
+the following ``BulkWriteModel`` classes:
+
+- ``BulkWriteInsertOneModel<TDocument>``
+- ``BulkWriteUpdateOneModel<TDocument>``
+- ``BulkWriteUpdateManyModel<TDocument>``
+- ``BulkWriteReplaceOneModel<TDocument>``
+- ``BulkWriteDeleteOneModel<TDocument>``
+- ``BulkWriteDeleteManyModel<TDocument>``
+
+The following sections show how to create and use instances of the preceding classes.
+
+.. tip:: Bulk Write Operations with POCOs
+
+    The examples in this guide use ``BsonDocument`` as the type for ``TDocument``.
+    You can also use a Plain Old CLR Object (POCO) as the type for ``TDocument``.
+    
+    To use a POCO, define a class that represents the documents in your collection.
+    The class must have properties that match the fields in your documents.
+    For more information, see :ref:`<csharp-poco>`.
+
+Insert Operations
+~~~~~~~~~~~~~~~~~
+
+To perform an insert operation, create an instance of the
+``BulkWriteInsertOneModel<TDocument>`` class.
+The ``BulkWriteInsertOneModel<TDocument>`` constructor accepts the following parameters:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Parameter
+     - Description
+   
+   * - ``collectionNamespace``
+     - | The database and collection to insert the BSON document into.
+       |
+       | **Data Type:** {+string-data-type+} or `CollectionNamespace <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.CollectionNamespace.html>`__
+
+   * - ``document``
+     - | The document to insert into the collection.
+       |
+       | **Data Type:** ``TDocument``
+
+The following example creates an instance of the ``BulkWriteInsertOneModel<TDocument>``
+class. The
+``BsonDocument`` object represents a new restaurant named "Mongo's Deli" to be inserted
+into the ``sample_restaurants.restaurants`` collection.
```

Comment:
Could you elaborate on what to avoid? I'm trying to think of another way to say it, because I'd normally use "model" in the context of data models; i.e., a POCO, or class that mirrors a specific document structure.

Reply:
I meant that we might prefer to avoid interchangeably referring to documents/C# types and restaurants. 
Looking back on my suggestion, it seems a bit clunky. Maybe something like:

> The ``BulkWriteInsertOneModel`` directs the driver to insert a document in which the ``name`` field value is ``"Mongo's Deli"`` into the ``restaurants`` collection.